pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7497,179 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
+	adminPasswordEnv := getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_ADMIN_PWD")
+	assert.Equal(t, len(adminUserEnv), 0)
+	assert.Equal(t, len(adminPasswordEnv), 0)
+}
+
+func TestCredentialWithCredentialSecretAndCredentials(t *testing.T) {
+	myNamespace := "myNamespace"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: myNamespace,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProduction,
+			CommonConfig: api.CommonConfig{
+				SecretAdminCredentials: constants.KIE_ADMIN_CREDENTIALS_SECRET,
+			},
+		},
+	}
+
+	service := test.MockService()
+	_, err := GetEnvironment(cr, service)
+	assert.Nil(t, err, "Error getting prod environment")
+
+	providedSecret := corev1.Secret{}
+	errS := service.Get(context.TODO(), types.NamespacedName{
+		Name:      constants.KIE_ADMIN_CREDENTIALS_SECRET,
+		Namespace: myNamespace,
+	}, &providedSecret)
+	assert.Nil(t, errS, "Error secret provided.")
+	adminUSer := providedSecret.StringData[constants.USERNAME_ADMIN_SECRET_KEY]
+	adminPassword := providedSecret.StringData[constants.PASSWORD_ADMIN_SECRET_KEY]
+	assert.True(t, len(adminUSer) > 0)
+	assert.True(t, len(adminPassword) > 0)
+	assert.Equal(t, adminUSer, constants.DefaultAdminUser)
+}
+
+func TestCredentialWithCredentialSecretAndCredentialsAlreadySet(t *testing.T) {
+	mySecretUsername := "mySecretUsername"
+	mySecretPassword := "mySecretPassword"
+	myNamespace := "myNamespace"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: myNamespace,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProduction,
+			CommonConfig: api.CommonConfig{
+				SecretAdminCredentials: constants.KIE_ADMIN_CREDENTIALS_SECRET,
+			},
+		},
+	}
+
+	service := test.MockService()
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.KIE_ADMIN_CREDENTIALS_SECRET,
+			Namespace: myNamespace,
+		},
+		Type: "Opaque",
+		StringData: map[string]string{
+			constants.USERNAME_ADMIN_SECRET_KEY: mySecretUsername,
+			constants.PASSWORD_ADMIN_SECRET_KEY: mySecretPassword,
+		},
+	}
+
+	errCreateSecret := service.Create(context.TODO(), &secret)
+	assert.Nil(t, errCreateSecret, "Error creating Secret")
+	_, err := GetEnvironment(cr, service)
+	assert.Nil(t, err, "Error getting prod environment")
+
+	providedSecret := corev1.Secret{}
+	errS := service.Get(context.TODO(), types.NamespacedName{
+		Name:      constants.KIE_ADMIN_CREDENTIALS_SECRET,
+		Namespace: myNamespace,
+	}, &providedSecret)
+
+	assert.Nil(t, errS, "Error secret created")
+	adminUserRetrieved := providedSecret.StringData[constants.USERNAME_ADMIN_SECRET_KEY]
+	adminPassword := providedSecret.StringData[constants.PASSWORD_ADMIN_SECRET_KEY]
+	assert.True(t, len(adminUserRetrieved) > 0)
+	assert.True(t, len(adminPassword) > 0)
+	assert.Equal(t, adminUserRetrieved, mySecretUsername)
+	assert.Equal(t, adminPassword, mySecretPassword)
```

Comment:
Please can you also add check that there is corectly set env variable for admin username and admin password for all deployments with used valzeForm secretKeyReference?

Reply:
No, if is in the env means no secret and legacy behaviour with username and password in cleartext